### PR TITLE
[refactor] merge duplicated parsing logic for queued operation

### PR DESCRIPTION
**Pull request by luxe**

### CODE REVIEW

File: src/main/java/build/buildfarm/common/ProtoUtils.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * @class ProtoUtils
+ * @brief Utilities related to parsing proto data.
+ * @details Performs validation and error reporting of proto data.
+ */
+public class ProtoUtils {
+  private static final Logger logger = Logger.getLogger(ProtoUtils.class.getName());
+
+  public static QueuedOperation getQueuedOperation(
+      ByteString queuedOperationBlob, QueueEntry queueEntry)
+      throws IOException, InterruptedException {
```
Comment (werkt) [inline src/main/java/build/buildfarm/common/ProtoUtils.java:36]: This should not be throwing either of these exceptions.

File: src/main/java/build/buildfarm/common/ProtoUtils.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    Digest queuedOperationDigest = queueEntry.getQueuedOperationDigest();
```
Comment (werkt) [inline src/main/java/build/buildfarm/common/ProtoUtils.java:37]: Pull `operation_name` here as well, if you're going to try to reduce interaction with queueEntry. You could also just take a string identifier for the pair and construct/pass it externally, but it's string construction that is only used in the error case.

**Review by werkt — CHANGES_REQUESTED** [review]

File: src/main/java/build/buildfarm/common/ProtoUtils.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+/**
+ * @class ProtoUtils
+ * @brief Utilities related to parsing proto data.
+ * @details Performs validation and error reporting of proto data.
+ */
+public class ProtoUtils {
+  private static final Logger logger = Logger.getLogger(ProtoUtils.class.getName());
+
+  public static QueuedOperation getQueuedOperation(
```
Comment (werkt) [inline src/main/java/build/buildfarm/common/ProtoUtils.java:33]: One last nit. This is now `parseQueuedOperation` by convention.

**Review by werkt — APPROVED** [review]

File: src/main/java/build/buildfarm/common/ProtoUtils.java (hunk 1 above)
Comment (luxe) [reply src/main/java/build/buildfarm/common/ProtoUtils.java]: fixed

File: src/main/java/build/buildfarm/common/ProtoUtils.java (hunk 2 above)
Comment (luxe) [reply src/main/java/build/buildfarm/common/ProtoUtils.java:35]: fixed

File: src/main/java/build/buildfarm/common/ProtoUtils.java (hunk 3 above)
Comment (luxe) [reply src/main/java/build/buildfarm/common/ProtoUtils.java]: fixed

**Pull request merged**